source/llassetgen/source/Image.cpp

```diff
@@ -72,15 +73,44 @@ namespace llassetgen {
         assert(ft_bitDepth == bitDepth);
 
         stride = size_t(std::ceil(float(getWidth() * bitDepth) / 8));
+        size_t paddingBytes = padding / 8;
+
+        /* because there are (padding % 8) padding bits, each byte from the original bitmap is
+           split into (8 - padding % 8) left bits and (padding % 8) right bits. Thus every byte
+           in the Image bitmap consists of the right bits of one original byte and the left bits
+           of the byte that comes after it.
+        */
+        char rightBits = padding % 8;
+        char leftBits = 8 - rightBits;
 
         data = std::shared_ptr<uint8_t>(new uint8_t[getHeight() * stride]);
-        for (size_t y = 0; y < getHeight(); y++) {
+
+        // y padding
+        for (size_t y = 0; y < padding; y++) {
             for (size_t x = 0; x < stride; x++) {
-                data.get()[y * stride + x] = 0xFF - ft_bitmap.buffer[y * ft_bitmap.pitch + x];
+                data.get()[y * stride + x] = 0xFF;
+                data.get()[(getHeight() - y - 1) * stride + x] = 0xFF;
+            }
+        }
+
+        for (size_t y = 0; y < ft_bitmap.rows; y++) {
```

Comment:
Why don't you use a view to offset the writing, isn't that precisely what they are meant for?